engine/src/flutter/impeller/renderer/backend/vulkan/descriptor_pool_vk.cc

```diff
@@ -187,15 +171,17 @@ vk::UniqueDescriptorPool DescriptorPoolRecyclerVK::Create() {
   return std::move(pool);
 }
 
-std::optional<vk::UniqueDescriptorPool> DescriptorPoolRecyclerVK::Reuse() {
-  Lock lock(recycled_mutex_);
-  if (recycled_.empty()) {
-    return std::nullopt;
+std::shared_ptr<DescriptorPoolVK>
+DescriptorPoolRecyclerVK::GetDescriptorPool() {
+  {
+    Lock recycled_lock(recycled_mutex_);
```

Comment:
Are we sure this is faster?  We are introducing locks.

Reply:
There were already locks. Before:

1. Raster: Finish work, push to resource manager thread.
2. Resource Manager: call reset, push to context (acquire lock).
3. Raster: Pick recycled pool.

Now we just skip the resource manager part, as we don't reset the pool unless it was completely unused.